test/e2e/specs/editor/various/a11y.spec.js

```diff
@@ -93,4 +111,87 @@ test.describe( 'a11y', () => {
 			)
 		).toBeFocused();
 	} );
+
+	test( 'should make the modal content focusable when it is scrollable', async ( {
+		page,
+	} ) => {
+		// Open the top bar Options menu.
+		await page.click(
+			'role=region[name="Editor top bar"i] >> role=button[name="Options"i]'
+		);
+
+		// Open the Preferences modal.
+		await page.click(
+			'role=menu[name="Options"i] >> role=menuitem[name="Preferences"i]'
+		);
+
+		const preferencesModal = page.locator(
+			'role=dialog[name="Preferences"i]'
+		);
+		const preferencesModalContent =
+			preferencesModal.locator( 'role=document' );
+		const closeButton = preferencesModal.locator(
+			'role=button[name="Close"i]'
+		);
+		const generalTab = preferencesModal.locator(
+			'role=tab[name="General"i]'
+		);
+		const blocksTab = preferencesModal.locator(
+			'role=tab[name="Blocks"i]'
+		);
+		const panelsTab = preferencesModal.locator(
+			'role=tab[name="Panels"i]'
+		);
+
+		// Check initial focus is on the modal dialog container.
+		await expect( preferencesModal ).toBeFocused();
+
+		// Check the General tab panel is visible by default.
+		await expect(
+			preferencesModal.locator( 'role=tabpanel[name="General"i]' )
+		).toBeVisible();
+
+		async function clickAndFocusTab( tab ) {
+			await tab.click();
+			await tab.focus();
```

Comment:
Why is this necessary? Wouldn't clicking automatically focus the tab?

Reply:
> Wouldn't clicking automatically focus the tab?

Not in all browsers, this behavior depends on the browser and the OS and it's actually the source of so many a11y bugs.

Reply:
I see, TIL! Then I think it's fine as long as it's not part of the user flow we want to test 👍 .

Reply:
Safari 😞 
I'll clarify with a comment.

Reply:
Safari 😞 
I'll clarify with a comment.